Tasks/NuGetCommandV2/nugetpublisher.ts

```diff
@@ -440,19 +446,30 @@ function getAccessToken(isInternalFeed: boolean): string{
             tl.debug("Checking for auth from Cred Provider."); 
             const feed = getProjectAndFeedIdFromInputParam('feedPublish');
             const JsonEndpointsString = process.env["VSS_NUGET_EXTERNAL_FEED_ENDPOINTS"];
+
             if (JsonEndpointsString) {
+                tl.debug(`Feed details ${feed.feedId} ${feed.projectId}`);
                 tl.debug(`Endpoints found: ${JsonEndpointsString}`);
 
                 let endpointsArray: { endpointCredentials: EndpointCredentials[] } = JSON.parse(JsonEndpointsString);
-                tl.debug(`Feed details ${feed.feedId} ${feed.projectId}`);
-
-                for (let endpoint_in = 0; endpoint_in < endpointsArray.endpointCredentials.length; endpoint_in++) {
-                    if (endpointsArray.endpointCredentials[endpoint_in].endpoint.search(feed.feedName) != -1) {
-                        tl.debug(`Endpoint Credentials found for ${feed.feedName}`);
-                        accessToken = endpointsArray.endpointCredentials[endpoint_in].password;
-                        break;
+                let matchingEndpoint: EndpointCredentials;
+
+                for (const e of endpointsArray.endpointCredentials) {
+                    // Only try the service connections that contain this org name
+                    if (e.endpoint.search(new RegExp(collectionPathName, "i")) != -1){
```

Comment:
Don't do this. This is treating collectionPathName as a regular expression. If collectionPathName contains characters that have special meanings in regexes, this will throw or behave unexpectedly.

Reply:
The caller of this method already has the list of prefixes as `uriPrefixes` at the point where this is called. Pass it in to this method and check whether any prefixes satisfy `e.endpoint.toUpperCase().startsWith(prefix.toUpperCase())`.